openedx/core/djangoapps/credit/tasks.py

```diff
@@ -33,23 +33,94 @@ def update_course_requirements(course_id):
         is_credit_course = CreditCourse.is_credit_course(course_key)
         if is_credit_course:
             course = modulestore().get_course(course_key)
-            requirements = [
-                {
-                    "namespace": "grade",
-                    "name": "grade",
-                    "criteria": {
-                        "min_grade": get_min_grade_for_credit(course)
-                    }
-                }
-            ]
+            requirements = _get_course_credit_requirements(course)
             set_credit_requirements(course_key, requirements)
-    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements) as exc:
+    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements, AttributeError) as exc:
         LOGGER.error('Error on adding the requirements for course %s - %s', course_id, unicode(exc))
         raise update_course_requirements.retry(args=[course_id], exc=exc)
     else:
         LOGGER.info('Requirements added for course %s', course_id)
 
 
-def get_min_grade_for_credit(course):
-    """ Returns the min_grade for the credit requirements """
-    return getattr(course, "min_grade", 0.8)
+def _get_course_credit_requirements(course):
+    """ Returns the list of credit requirements for the given course
+
+    It returns the minimum_grade_credit and also the ICRV checkpoints
+    if added any in the course
+
+    Args:
+        course(Course): The course object
+
+    Returns:
+        List of minimum_grade_credit and ICRV requirements
+    """
+    icrv_requirements = _get_credit_course_requirement_xblocks(course)
+    min_grade_requirement = _get_min_grade_requirement(course)
+    icrv_requirements.extend(min_grade_requirement)
```

Comment:
Nitpick: I think you could make this just a tiny bit more readable by using the `+=` operator. Also, FYI, extending `icrv_requirements` with `extend` involves a function call which the `+=` does not.